Enable turbopackFileSystemCacheForBuild by default (#96395)
vercel/next.js: docs/01-app/03-api-reference/05-config/01-next-config-js/turbopackFileSystemCache.mdx

Turbopack FileSystem Cache enables Turbopack to reduce work across `next dev` or `next build` commands. When enabled, Turbopack will save and restore data to the `.next` folder between builds, which can greatly speed up subsequent builds and dev sessions.

> > **Good to know:** The FileSystem Cache is **enabled by default** for both `next dev` and `next build`. For `next build` it is disabled by default in non-Vercel CI environments, where the cache is unlikely to persist between builds. You can explicitly enable or disable it with the flags below.

Reviewer: Avoid the negatives.

```suggestion
> **Good to know:** The FileSystem Cache is **enabled by default** for `next dev`. It is enabled by default when running `next build` locally or on Vercel, where the cache is likely to persist between builds. You can explicitly enable or disable it with the flags below.
```

Can we also put a note here about what environment variables other providers can set to opt-into enabling this by default?

cc @icyJoseph

## Version Changes

| Version   | Changes                                                                                    |
| --------- | ------------------------------------------------------------------------------------------ |

> | `v16.3.0` | FileSystem caching is enabled by default for builds (disabled by default in non-Vercel CI) |

Reviewer: Make this less negative and less Vercel-specific

```suggestion
| `v16.3.0` | FileSystem caching is enabled by default for builds (depends on CI platform support) |
```
